proguard/src/main/java/com/flurry/proguard/UploadProguardMapping.java

```diff
@@ -0,0 +1,216 @@
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.FileEntity;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.GZIPOutputStream;
+
+public class UploadProguardMapping {
+
+    private static final String METADATA_BASE = "https://crash-metadata.flurry.com/pulse/v1/";
+    private static final String UPLOAD_BASE = "https://upload.flurry.com/upload/v1/upload/";
+    private static HttpClient httpClient;
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        if (args.length != 3) {
+            System.out.println("Please provide 3 arguments in the following order:");
+            System.out.println("ApiKey PathToProguard AccessToken");
+            throw new IOException("Invalid number of arguments");
+        }
+        httpClient = HttpClientBuilder.create().build();
+        new UploadProguardMapping().uploadFile(args[0], args[1], args[2]);
+    }
+
+    public void uploadFile(String apiKey, String pathToFiles, String token)
+            throws IOException, InterruptedException {
+        File file = new File(pathToFiles);
+        File zippedFile = tarZipFile(file);
+        String projectId = lookUpProjectId(apiKey, token);
+        String payload = getJson("/json/upload.json").replace("UPLOAD_SIZE",
+                Long.toString(zippedFile.length()))
+                .replace("PROJECT_ID", projectId);
+        String uploadId = createUpload(projectId, payload, token);
+        System.out.println("Upload created with upload ID: " + uploadId);
+        sendToUploadService(zippedFile, projectId, uploadId, token);
+        String retval = checkUploadStatus(projectId, uploadId, token);
+        if (retval != null) {
+            throw new IOException("Failed to upload with Failure Reason: " + retval);
+        }
+        System.out.println("SUCCESS!!");
+    }
+
+    private File tarZipFile(File file) throws IOException {
+        File tarZippedFile = File.createTempFile("tar-zipped-file", ".tar.gz");
+        TarArchiveOutputStream taos = new TarArchiveOutputStream(new GZIPOutputStream(new BufferedOutputStream(
+                new FileOutputStream(tarZippedFile))));
+        putArchives(taos, file, "");
+        taos.finish();
+        taos.close();
+        return tarZippedFile;
+    }
+
+    private void putArchives(TarArchiveOutputStream taos, File file, String base) throws IOException {
+        if (file.isDirectory()) {
+            for (File f : file.listFiles()) {
```

Comment:
Instead of using recursion, just use FileUtils which will do the same - and allow this code to stay iterative.